jetty-http/src/main/java/org/eclipse/jetty/http/pathmap/PathSpec.java

```diff
@@ -25,6 +25,17 @@
  */
 public interface PathSpec extends Comparable<PathSpec>
 {
+    static PathSpec from(String pathSpecString)
+    {
+        if (pathSpecString == null)
+            throw new RuntimeException("Path Spec String must start with '^', '/', or '*.': got [" + pathSpecString + "]");
```

Comment:
Should not this be a NPE? 
Furthermore, if it's non-null, there is no check that `pathSpecString` starts with ^, /, etc. The only check below is for ^ to distinguish between regex or not.